### ISSUE TITLE
Source Object Validation Invalid Results if Session Exists

### ISSUE THREAD

**Issue by pke11y** [nautobot/nautobot-plugin-circuit-maintenance#77]
### Environment

- Python version: 3.8
- Nautobot version: 1.0.3
- nautobot-circuit-maintenance version: 0.1.7

When validating a notification source using the `test_authentication` method on the CLI, if a session already exists the return method for the test will always be `True - Login OK`

### Expected Behavior
Ignore session data when testing the notification source. It may be required to change attributes in the source during initial setup and source validation.

### Observed Behavior

Running `test_authentication()` will always return True after initial attempt as the session data causes a conditional check in the logic to avoid subsequent connection attempts.

### Steps to Reproduce

1. Open CLI session and import nautobot
2. `from nautobot_circuit_maintenance.handle_notifications.sources import Source`
3. Create a test source and execute `test_authentication`
```
from nautobot_circuit_maintenance.handle_notifications.sources import Source
>>> source = Source.init(name="TEST IMAP")
>>> source.test_authentication()
(False, 'LOGIN failed.')
>>> source.test_authentication()
(True, 'Test OK')
```

**Comment by pke11y** [comment]
#76 aimed at solving this issue

**Comment by chadell** [comment]
Fixed in #76
